ios/brave-ios/Sources/Brave/Frontend/Shields/ShieldsPanelView.swift

```diff
@@ -0,0 +1,413 @@
+// Copyright 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import BraveCore
+import BraveShields
+import BraveUI
+import Data
+import DesignSystem
+import Favicon
+import SnapKit
+import Strings
+import SwiftUI
+
+struct ShieldsPanelView: View {
+  enum Action {
+    enum NavigationTarget {
+      case shareStats
+      case reportBrokenSite
+      case globalShields
+    }
+    case navigate(NavigationTarget, dismiss: Bool)
+    case changedShieldSettings
+    case shredSiteData
+  }
+
+  private let url: URL
+  private let displayHost: String
+  @State private var favicon = Image(uiImage: Favicon.defaultImage)
+  @AppStorage("advancedShieldsExpanded") private var advancedShieldsExpanded = false
+  @ObservedObject private var viewModel: ShieldsSettingsViewModel
+  private var actionCallback: (Action) -> Void
+
+  @MainActor init(url: URL, tab: Tab, domain: Domain, callback: @escaping (Action) -> Void) {
+    self.url = url
+    self.viewModel = ShieldsSettingsViewModel(tab: tab, domain: domain)
+    self.actionCallback = callback
+    self.displayHost = URLFormatter.formatURLOrigin(
+      forDisplayOmitSchemePathAndTrivialSubdomains: url.absoluteString
+    )
+  }
+
+  private var shieldsEnabledAccessibiltyLabel: String {
+    viewModel.shieldsEnabled
+      ? Strings.Shields.statusValueUp : Strings.Shields.statusValueDown
+  }
+
+  var body: some View {
+    ScrollView {
+      VStack(spacing: 16) {
+        headerView
+
+        if viewModel.shieldsEnabled {
+          shieldsReportView
+          Text(Strings.Shields.siteBroken)
+            .font(.caption)
+            .foregroundStyle(Color(.secondaryBraveLabel))
+            .multilineTextAlignment(.leading)
+            .padding(.horizontal)
+          DisclosureGroup(isExpanded: $advancedShieldsExpanded) {
+            advancedShieldsSection
+          } label: {
+            Text(Strings.Shields.advancedControls)
+              .foregroundStyle(Color(.braveLabel))
+              .frame(maxWidth: .infinity, alignment: .leading)
+          }.disclosureGroupStyle(ShieldsPanelDisclosureStyle())
+        } else {
+          shieldsOffFooterView
+        }
+      }
+      .padding(.top)
+    }
+    .background(Color(.braveBackground))
+    .frame(idealWidth: 360, alignment: .center)
+    .task {
+      if let faviconImage = FaviconFetcher.getIconFromCache(for: url)?.image {
+        self.favicon = Image(uiImage: faviconImage)
+      }
+      await loadFavicon(for: url, isPrivateBrowsing: viewModel.isPrivateBrowsing)
```

Comment:
There is a favicon SwiftUI view already. Use that. Wallet and AI-Chat both use it.

Reply:
Perfect, thanks!